dace/codegen/sdfgcc.py

```diff
@@ -4,23 +4,47 @@
 import dace
 import os
 import sys
+import argparse
+import shutil
 
 
 def main():
-    if len(sys.argv) != 2:
-        print('USAGE: sdfgcc <PATH TO SDFG FILE>')
+    # Command line options parser
+    parser = argparse.ArgumentParser(
+        description='Simple SDFG command-line compiler.')
+
+    # Required argument for SDFG file path
+    parser.add_argument('filepath', help='<PATH TO SDFG FILE>', type=str)
+
+    # Optional argument for output location
+    parser.add_argument(
+        '-o',
+        '--out',
+        type=str,
+        help=
+        'If provided, saves lib and header file to path. Directories in path need to exist beforehand.'
+    )
+
+    args = parser.parse_args()
+
+    filepath = args.filepath
+    if not os.path.isfile(filepath):
+        print('SDFG file', filepath, 'not found')
         exit(1)
 
-    filename = sys.argv[1]
-    if not os.path.isfile(filename):
-        print('SDFG file', filename, 'not found')
-        exit(2)
+    outpath = args.out
 
     # Load SDFG
-    sdfg = dace.SDFG.from_file(filename)
+    sdfg = dace.SDFG.from_file(filepath)
 
     # Compile SDFG
-    sdfg.compile()
+    sdfg.compile(outpath)
+
+    # Copying header file to optional path
+    if args.out:
+        source = os.path.join(sdfg.build_folder, 'src', 'cpu', sdfg.name + '.h')
+        destination = os.path.join(os.path.dirname(outpath), sdfg.name + '.h')
```

Comment:
That won't work if a path is given and does not have a trailing slash. Try using `os.path.isdir` instead